pkg/server/endpoints/endpoints_test.go

```diff
@@ -197,32 +199,37 @@ func (s *EndpointsTestSuite) TestHTTPServerConfig() {
 }
 
 func (s *EndpointsTestSuite) TestSVIDObserver() {
-	state := observer.NewProperty(svid.State{})
-	s.e.c.SVIDStream = state.Observe()
-
 	ctx, cancel := context.WithCancel(ctx)
-	t := new(tomb.Tomb)
-	defer func() {
-		cancel()
-		s.Require().NoError(t.Wait())
+	defer cancel()
+
+	go func() {
+		s.e.runSVIDObserver(ctx)
 	}()
-	t.Go(func() error {
-		return s.e.runSVIDObserver(ctx)
-	})
-
-	cert, key, err := util.LoadSVIDFixture()
-	s.Require().NoError(err)
-	svid := svid.State{
-		SVID: cert,
-		Key:  key,
-	}
-	state.Update(svid)
 
-	time.Sleep(1 * time.Millisecond)
-	s.e.mtx.RLock()
-	defer s.e.mtx.RUnlock()
-	s.Assert().Equal(cert, s.e.svid)
-	s.Assert().Equal(key, s.e.svidKey)
+	expectedState := svid.State{
+		SVID: &x509.Certificate{},
```

Comment:
I would need a bit of help here to understand why it doesn't use the cert and key returned by util.LoadSVIDFixture() any more

Reply:
It didn't actually use the certificate as part of the test, just makes sure that value is pulled out of the property when it changes, so it didn't seem useful to load up the certificate for no reason. It also doesn't hurt anything to continue load the cert from disk, so if you feel strongly not to change it i'm happy to put it back.

I will however push an update that has clearer documentation in the test case.

Reply:
I see now, it is fine as is. Thanks for the docs and clarification!